sklearn/datasets/_arff_parser.py

```diff
@@ -371,11 +371,27 @@ def _pandas_arff_parser(
         names=[name for name in openml_columns_info],
         dtype=dtypes,
     )
-
     columns_to_select = feature_names_to_select + target_names_to_select
     columns_to_keep = [col for col in frame.columns if col in columns_to_select]
     frame = frame[columns_to_keep]
 
+    # strip quotes to be consistent with LIAC ARFF
+    re_start_end_quotes = "^[\"'](.*[\"'])$"
+
+    def strip_quotes(s):
+        return s.group(0)[1:-1]
+
+    for cat_col in frame.select_dtypes(include="category").columns:
```

Comment:
Do you think it is a little safer to use `frame.dtypes`? I'm unsure if `select_dtypes` makes a copy.